Dockerfile

```diff
@@ -134,7 +134,7 @@ RUN \
 ADD config /etc/nginx/
 ADD *.sh /usr/bin/
 
-VOLUME /etc/nginx/conf.d /var/log/nginx
+VOLUME /etc/nginx/conf.d /etc/nginx/lua /var/log/nginx
```

Comment:
It should be `/etc/nginx/canary` instead of `/etc/nginx/lua`. The directory `/etc/nginx/lua` has symlinks to `/etc/nginx/canary` for versions.lua and config.lua, but it also contains other scripts with the real logic for canary deployment strategy.